llvm/include/llvm/CodeGen/MachineValueType.h

```diff
@@ -33,7 +33,7 @@ namespace llvm {
   /// type can be represented by an MVT.
   class MVT {
   public:
-    enum SimpleValueType : uint8_t {
+    enum SimpleValueType : uint16_t {
```

Comment:
This is doubling the size of MVT, right? Is there any measurable impact on compilation speed or memory usage?

Reply:
I have not done any measurements. Do you have a suggestion as to what experiment you'd like me to do?

Reply:
Maybe start with creating a perf/<branchname> branch and asking @nikic to add you to https://llvm-compile-time-tracker.com/ ?

Reply:
Interesting! Thank you. I'll do that.